app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,40 @@
+class SessionsController < ApplicationController
+  before_action :find_user, only: [:create]
+
+  # GET /login
+  def new; end
+
+  def create
+    if @user && authenticate_user(@user)
+      handle_success(@user)
+    else
+      handle_failure
+    end
+  end
+
+  def destroy
+    log_out if logged_in?
+    flash[:success] = t("flash.logged_out")
+    redirect_to root_path, status: :see_other
+  end
+  private
+  def find_user
+    @user = User.find_by(email: params[:session][:email].downcase)
```

Comment:
```suggestion
    @user = User.find_by(email: params.dig(:session, :email)&.downcase)
```
Dùng dig và toán tử an toàn để tránh crash áp